profiles/killbill/src/test/java/org/killbill/billing/server/log/obfuscators/TestPatternObfuscator.java

```diff
@@ -17,16 +17,112 @@
 
 package org.killbill.billing.server.log.obfuscators;
 
-import org.killbill.billing.server.log.ServerTestSuiteNoDB;
+import java.util.Collection;
+import java.util.List;
+
 import org.mockito.Mockito;
 import org.testng.Assert;
+import org.testng.annotations.BeforeClass;
 import org.testng.annotations.Test;
 
 import ch.qos.logback.classic.spi.ILoggingEvent;
 
-public class TestPatternObfuscator extends ServerTestSuiteNoDB {
+public class TestPatternObfuscator {
+
+    private PatternObfuscator obfuscator;
+
+    @BeforeClass(groups = "fast")
+    public void beforeClass() throws Exception {
+        System.setProperty("killbill.server.log.obfuscate.keywords", "req.requestURI,accountnumber,authenticationdata,banknumber,bic,cardvalidationnum,cavv,ccFirstName,ccLastName,ccNumber,ccTrackData,ccVerificationValue,ccvv,cvNumber,cvc,cvv,email,iban,name,number,password,xid,cell_phone");
```

Comment:
I would suggest to only have the shared properties for all tests here: If those are test dependent, then they should probably be set inside the test itself, and cleanup afterwards.

Reply:
The System properties need to be set before invoking the `PatternObfuscator` constructor as the constructor then loads the keywords/patterns [here](https://github.com/reshmabidikar/killbill/blob/916d4a2454a2b36084d522f69289c0493651adbb/profiles/killbill/src/main/java/org/killbill/billing/server/log/obfuscators/PatternObfuscator.java#L53). If we set the properties in the test, we would need to create a new instance of the obfuscator in the test after the properties are set. 

So, the test would need to be modified as follows:
```
    public void testJSONWithCustomKeyWord() throws Exception {
        System.setProperty("killbill.server.log.obfuscate.keywords","accountnumber,authenticationdata,banknumber,bic,cardvalidationnum,cavv,ccFirstName,ccLastName,ccNumber,ccTrackData,ccVerificationValue,ccvv,cvNumber,cvc,cvv,email,iban,name,number,password,xid,cell_phone");
        PatternObfuscator obfuscator = new PatternObfuscator();
        verify(obfuscator,"{\n" +
               "  \"card\": {\n" +
               "    \"id\": \"card_483etw4er9fg4vF3sQdrt3FG\",\n" +
               "    \"object\": \"card\",\n" +
               "    \"banknumber\": 4111111111111111,\n" +
               "    \"cvv\" : 111,\n" +
               "    \"cvv\": 111,\n" +
               "    \"cvv\": \"111\",\n" +
               "    \"data\": {\"cvv\" : 111 },\n" +
               "    \"last4\": \"0000\",\n" +
               "    \"brand\": \"Visa\",\n" +
               "    \"funding\": \"credit\",\n" +
               "    \"exp_month\": 6,\n" +
               "    \"exp_year\": 2019,\n" +
               "    \"fingerprint\": \"HOh74kZU387WlUvy\",\n" +
               "    \"country\": \"US\",\n" +
               "    \"name\": \"Bob Smith\",\n" +
               "    \"cell_phone\": \"212-421-1278\",\n" +
               "    \"address_line1\": null,\n" +
               "    \"address_line2\": null,\n" +
               "    \"address_city\": null,\n" +
               "    \"address_state\": null,\n" +
               "    \"address_zip\": null,\n" +
               "    \"address_country\": null,\n" +
               "    \"dynamic_last4\": \"4242\",\n" +
               "    \"customer\": null,\n" +
               "    \"type\": \"Visa\"}\n" +
               "}",
               "{\n" +
               "  \"card\": {\n" +
               "    \"id\": \"card_483etw4er9fg4vF3sQdrt3FG\",\n" +
               "    \"object\": \"card\",\n" +
               "    \"banknumber\": ****************,\n" +
               "    \"cvv\" : ***,\n" +
               "    \"cvv\": ***,\n" +
               "    \"cvv\": *****,\n" +
               "    \"data\": {\"cvv\" : ****},\n" +
               "    \"last4\": \"0000\",\n" +
               "    \"brand\": \"Visa\",\n" +
               "    \"funding\": \"credit\",\n" +
               "    \"exp_month\": 6,\n" +
               "    \"exp_year\": 2019,\n" +
               "    \"fingerprint\": \"HOh74kZU387WlUvy\",\n" +
               "    \"country\": \"US\",\n" +
               "    \"name\": ***********,\n" +
               "    \"cell_phone\": **************,\n" +
               "    \"address_line1\": null,\n" +
               "    \"address_line2\": null,\n" +
               "    \"address_city\": null,\n" +
               "    \"address_state\": null,\n" +
               "    \"address_zip\": null,\n" +
               "    \"address_country\": null,\n" +
               "    \"dynamic_last4\": \"4242\",\n" +
               "    \"customer\": null,\n" +
               "    \"type\": \"Visa\"}\n" +
               "}");

    }
```
If you agree, I'll make this change for all the tests, let me know what you think.

Reply:
If you look at your existing tests, they already set some system properties within the test. For example in `testLoadKeywords`, I see:

```
    @Test
    public void testLoadKeywords() {
        //without setting the killbill.server.log.obfuscate.patterns property
        System.clearProperty("killbill.server.log.obfuscate.keywords");
        Collection<String> keywords = obfuscator.loadKeywords();

        ...

        System.setProperty("killbill.server.log.obfuscate.keywords", "accountnumber,authenticationdata");
        keywords = obfuscator.loadKeywords();
        Assert.assertEquals(keywords.size(), 2);
```


Anyhow, it's really your choice if you want to reuse the same `obfuscator` instance through the tests or create a new one each time. All I am asking is 1/ consistency across the tests, 2/ proper logic to make sure we clean things up after each tests to avoid weird behavior on subsequent tests and ideally 3/ set the minimum we need.

Reply:
The `testLoadKeywords` test is different from the other tests in that it loads the keywords within the test and is not concerned about when the `PatternObfuscator` instance is created.

I've anyhow modified the tests to set the system property within the test and create an `obfuscator` instance within the test after setting the property.  I've also added an `@AfterMethod` to clear the properties after each test, so hopefully this is good now.